tools/ncf_constraints.py

```diff
@@ -0,0 +1,107 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+if sys.version_info < (3,):
+  string_type = unicode
+else:
+  string_type = str
+
+### Constraint checking function
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return len(parameter) <= max_size
+
+def min_length( parameter, min_size):
+  return len(parameter) >= min_size
+
+def match_regex(parameter, regex):
+  match = re.search(regex, parameter)
+  return match is not None
+
+def not_match_regex(parameter, regex):
+  return not match_regex(parameter, regex)
+
+def allow_empty_string(parameter, allow):
+  return allow or not_match_regex(parameter, "^$")
+
+def allow_whitespace_string(parameter, allow):
+  return allow or not_match_regex(parameter, r'^\s') or not_match_regex(parameter, r'\s$')
+
+
+constraints = {
+    "list" : {
+        "check" : from_list
+      , "type"  : list
+    }
```

Comment:
Re-reading the email discussion we had about specs makes me notice the name you've used here: "list". I don't think this is a good name, because it is not clear what "constraint" "list" is imposing.

I prefer "enum" (which, to me at least, makes it clear that this constraint in an enumeration of possible values), but I'm open to other terms. Maybe "values"?

Thoughts?